crates/common_utils/src/lib.rs

```diff
@@ -33,6 +33,15 @@ pub mod date_time {
     pub fn now_unix_timestamp() -> i64 {
         OffsetDateTime::now_utc().unix_timestamp()
     }
+
+    /// Calculate execution time for a async block in milliseconds
+    pub async fn time_it<T, Fut: futures::Future<Output = T>, F: FnOnce() -> Fut>(
+        block: F,
+    ) -> (T, f64) {
+        let start = Instant::now();
+        let result = block().await;
+        (result, start.elapsed().as_seconds_f64() * 1000f64)
```

Comment:
If you need milliseconds as an integer, you could also use `as_millis()`.

`(result, start.elapsed().as_millis())`

Reply:
Here we want the time in f64 milliseconds, as_millies returns u128 which will not preserve the precision. That's the reason I used seconds as f64